rpcperms/interceptor.go

```diff
@@ -751,6 +778,22 @@ func (r *InterceptorChain) checkRPCState(srv interface{}) error {
 
 		return ErrRPCStarting
 
+	// If lnd is waiting for the remote signer to connect, we only allow
+	// calls to the remote signer.
+	case allowRemoteSigner:
+		_, ok := srv.(lnrpc.WalletUnlockerServer)
+		if ok {
+			return ErrWalletUnlocked
+		}
+
+		// As we only allow calls to connect the remote signer until the
+		// full rpc server is active, we check whether the method is
+		// whitelisted or not.
+		_, ok = allowRemoteSignerWhitelist[fullMethod]
```

Comment:
Still going through the diff, but I think we should add an itest specifically for testing that we cannot access this method without a valid macaroon (since we're changing the logic of the permission interceptor here).

Reply:
Good idea! I added a separate `testOutboundRSMacaroonEnforcement` itest for that :)